### ISSUE TITLE
Statement Generator Account List Does not Honor Rock Account Order

### ISSUE THREAD

**Issue by rutledgek** [SparkDevNetwork/Rock#4119]
### Description
Submitted on Behalf of The Woodlands UMC

### Steps to Reproduce
1. Go to Financial Accounts and Note Sort Order
1. Open Statement Generator 
1. View All Places that List Accounts
1. Accounts are in Alphabetical Order rather than Order established in Rock

**Expected behavior:**

That the account order set in the Financial Accounts would be the default wherever financial accounts are shown/used including the Statement Generator

**Actual behavior:**

Account order is not respected in various places around Rock including the Statement Generator.

### Versions
Rock Version: Rock McKinley 9.4 (1.9.4.1)
Client Culture Setting: en-US

**Comment by nairdo** [comment]
@rutledgek We've attempted to reproduce this without success.  We've asked the beta testers to attempt and they'll update here if they find anything.

**Comment by smross** [comment]
@rutledgek Can you provide some more insight into the accounts that you're seeing this happen for?

*TL;DR* I can't prove this is an issue we experience. I don't see how it could be consistent, as the Rock UI allows for parent/child relationships, and the statement generator UI doesn't have a mechanism for that in my case. i'm not sure what's expected..... we just live with it, and without showing the tree in Statement Generator, I don't know how it would be a 'mirror' of the main Rock UI 

In our case we have a bit of a hierarchy for organization purposes. So, it's not really clear how the 'order' would translate, as the Statement Generator doesn't have a tree in use (like we do).
This may or may not be confirmation, but here's what I experience:
* We have 3 top-level Accounts as grouping mechanisms: "Tithe & Offering", "Tithe & Offering Cash", and "Budget Accounts". We do not have any top-level accounts besides these grouping accounts
* The top-level accounts may have grouping accounts, or actual accounts as children. The _Budget Accounts_ top-level has a lot of child accounts with child accounts themselves
* The first 2 accounts for _Tithe & Offering_ are "1001C - St. Charles...." and "1001M - mid Rivers"
* The first 2 accounts for _Tithe & Offering Cash_ are "1001C - St. Charles" and "1001M - Mid Rivers"
* When I open Statement Generator and open last year (to get the largest amount of accounts with data). I have selected _Show Tax Deductible Accounts_ and _Show Non Tax Deductible Accounts_ selected, along with everything but non-cash for the "Currency types for cash gifts" section, and only non-cash asset for the "Currency types for cash gifts" section. I see:
** Accounts with a preceding # first. If the accounts have the same preceding sequence (i.e. 3 with _1001C_ in the front of their name), then *_those accounts are alphabetized_* (but they come from different tree sections, so i don't know what is the expected behavior).
** It looks like _sections_ may be alphabetized, but the whole thing is not alphabetized

Pictures showing what I experience...
<img width="483" alt="Accounts | Rock RMS 2020-03-01 14-42-10" src="https://user-images.githubusercontent.com/8688424/75633553-5291d880-5bcb-11ea-9754-de295db74de7.png">
![Screen Shot 2020-03-01 at 2 50 14 PM](https://user-images.githubusercontent.com/8688424/75633668-8f120400-5bcc-11ea-9cf0-de6257906495.png)


My system I'm testing on info:
* Version: 9.4 (1.9.4.1)
* Client Culture: en-US

**Comment by rutledgek** [comment]
This is based on a client request.  I am torn on which is better.  This list appears to be global alphabetical which in makes sense.  However, I can also see the value of a Parent -> Children -> subparent->children list.  On the one hand, this would require knowledge of the tree structure in rock, on the other, the tree structure usually matches up with account groups.  Tithe groups together, event groups together, campuses divided, etc so seeing them all grouped together as they are in the tree could be helpful.

**Comment by smross** [comment]
I think in a perfect world the UI in the Statement Generator would match Rock's UI. But honestly we only use the statement generator a small # of times a year (not daily). Therefore, I have no complaints. Does your client have feedback on how much of an impact this is having?

I disagree with you on the sorting. I don't see evidence that it is alpha sorted in the Statement Generator. I don't know how it's sorted, but my 2nd screenshot shows that a portion of my list is _not_ alpha sorted

**Comment by nairdo** [comment]
Since the Statement Generator has gotten a major overhaul in recent releases, I'm pretty certain this issue has been resolved as of Rock v12.5.  Nowadays, the accounts are selected during the creation of the 'Contribution Templates' under _Financial Settings_.

![image](https://user-images.githubusercontent.com/383807/167230590-85009bb1-7195-4378-adbd-7970f578ce37.png)

https://rock.rocksolidchurchdemo.com/finance/settings/contribution-templates/1
![image](https://user-images.githubusercontent.com/383807/167230602-bef08c51-dfbc-4103-a66d-e9c30f72268c.png)

Then, during step 2 when running the Statement Generator app, you choose the template that has the configuration you want.

![image](https://user-images.githubusercontent.com/383807/167230642-efcffdf3-9b5a-4c25-b630-84ed6b5d2db6.png)
